crates/router/src/services/api.rs

```diff
@@ -563,16 +565,26 @@ where
 pub async fn call_connector_api(
     state: &AppState,
     request: Request,
+    flow_name: &str,
 ) -> CustomResult<Result<types::Response, types::Response>, errors::ApiClientError> {
     let current_time = Instant::now();
-
+    let headers = request.headers.clone();
+    let url = request.url.clone();
     let response = state
         .api_client
         .send_request(state, request, None, true)
         .await;
 
-    let elapsed_time = current_time.elapsed();
-    logger::info!(request_time=?elapsed_time);
+    let _ = response.as_ref().map(|resp| {
+        let status_code = resp.status().as_u16();
+        let elapsed_time = current_time.elapsed();
+        logger::info!(
+            headers=?headers,
+            url=?url,
+            status_code=?status_code,
+            flow=?flow_name,
+            elapsed_time=?elapsed_time)
+    });
```

Comment:
Nit:

If we're not returning anything, I'd prefer using `if let Some(resp) = response.as_ref() { ... }` over `map()`.